src/PluginStartupRegistration.cpp

```diff
@@ -34,20 +35,44 @@ void PluginStartupRegistration::OnInternalError(const wxString& error)
 
 void PluginStartupRegistration::OnPluginFound(const PluginDescriptor& desc)
 {
-   PluginManager::Get().RegisterPlugin(PluginDescriptor { desc });
-   if(!desc.IsValid())
```

Comment:
There is no longer a check of IsValid().  The condition for considering a plug-in failed is instead that all descriptors for the path are stubs.

I had to study the code again to be convinced that the condition for considering a plug-in failed, in case of only one provider, is equivalent to the old criterion.  But, I believe it now.

Plugin destriptors deserialized from the IPC pipe are stubs if and only if they are marked invalid.  Otherwise they were successfully registered in PluginManager in the other process and marked valid.